ext/nginx.conf.example

```diff
@@ -0,0 +1,85 @@
+# NOTE: if you wish to only connect using fallback, you can
+# remove all ssl related content from the configuration.
+
+# c[e4]?.ppy.sh is used for bancho
+# osu.ppy.sh is used for /web, /api, etc.
+# a.ppy.sh is used for osu! avatars
+
+server {
+	# Use port 443 if you aren't going to use Cloudflare (flex)
+	# and prefer to use your own certificates.
+	listen 80;
+	#listen 443 ssl;
+
+	# XXX: you'll need to edit this to match your domain
+	server_name ~^(?:c[e4]?|osu|b|api)\.example\.com$;
+
+	# Uncomment these paths if you aren't going to use Cloudflare (flex)
+	# Both are hardcoded in docker-compose.yml so now you
+	# don't need to change these unless you changed that.
+	#ssl_certificate     /etc/ssl/certs/bancho.pem;
+	#ssl_certificate_key /etc/ssl/private/bancho.key;
+
+	# TODO: further ssl configuration
+	# Uncomment this if you aren't going to use Cloudflare (flex)
+	#ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
+
+	location / {
+		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+		proxy_set_header X-Real-IP  $remote_addr;
+		proxy_set_header Host $http_host;
+		add_header Access-Control-Allow-Origin *;
+		proxy_redirect off;
+		proxy_pass http://bancho;
+	}
+}
+
+server {
+	# Use port 443 if you aren't going to use Cloudflare (flex)
+	# and prefer to use your own certificates.
+	listen 80;
+	#listen 443 ssl;
+
+	# XXX: you'll need to edit this to match your domain
+	server_name assets.example.com;
+
+	# Uncomment these paths if you aren't going to use Cloudflare (flex)
+	# Both are hardcoded in docker-compose.yml so now you
+	# don't need to change these unless you changed that.
+	#ssl_certificate     /etc/ssl/certs/bancho.pem;
+	#ssl_certificate_key /etc/ssl/private/bancho.key;
+
+	# TODO: further ssl configuration
+	# Uncomment this if you aren't going to use Cloudflare (flex)
+	#ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
+
+	location / {
+		default_type image/png;
+		root /.data/assets;
+	}
+}
+
+server {
+	# Use port 443 if you aren't going to use Cloudflare (flex)
+	# and prefer to use your own certificates.
+	listen 80;
+	#listen 443 ssl;
+
+	# XXX: you'll need to edit this to match your domain
+	server_name a.example.com;
+
+	# Uncomment these paths if you aren't going to use Cloudflare (flex)
+	# Both are hardcoded in docker-compose.yml so now you
+	# don't need to change these unless you changed that.
+	#ssl_certificate     /etc/ssl/certs/bancho.pem;
+	#ssl_certificate_key /etc/ssl/private/bancho.key;
+
+	# TODO: further ssl configuration
+	# Uncomment this if you aren't going to use Cloudflare (flex)
+	#ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
+
+	location / {
+		root /.data/avatars;
+		try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
+	}
+}
```

Comment:
this file is getting a bit complex, i wonder if we can make any simplifications.. might try this myself

Reply:
bump for @cmyui